js/app.js

```diff
@@ -317,8 +320,38 @@ document.querySelector('#contact-form').addEventListener('submit', (e) => {
     }
   }
 
-  if (!valid) {
+  // if the email is valid, save user data in object
+  if (valid) {
+    // prevent the form fro automatically reloading the page
+    e.preventDefault();
+
+    // empty the response test in contact form
+    responseContainer.innerHTML = '';
+    // save the user info to object
+    const details = {
+      name,
+      email,
+      message,
+    };
+
+    // save the user object to localStorage
+    // (here we use JSON.stringify() to convert the user data to string for safe storage)
+    localStorage.setItem('contact-details', JSON.stringify(details));
+  } else {
     e.preventDefault();
     responseContainer.innerHTML = 'Invalid Email';
   }
 });
+
+// function to populate the contact form from localStorage
+function loadFormData() {
+  if (localStorage.getItem('contact-details') != null) {
+    const userData = JSON.parse(localStorage.getItem('contact-details'));
+
+    document.querySelector('#email').value = userData.email;
+    document.querySelector('#contact-name').value = userData.name;
+    document.querySelector('#contact-message').value = userData.message;
+  }
+}
```

Comment:
- [ ] Kindly when I submitted the data saved in the local storage but when I reload the page the data is not appearing.

![image](https://user-images.githubusercontent.com/35535130/156573129-312e4512-00b0-4f9f-8b06-430bec86521c.png)